src/services/earn/earn-service.ts

```diff
@@ -0,0 +1,262 @@
+import { Address, BigIntish, BuiltTransaction, ChainId, TokenAddress } from '@types';
+import { CreateEarnPositionParams, EarnActionSwapConfig, EarnPermission, IEarnService } from './types';
+import { COMPANION_SWAPPER_CONTRACT, EARN_STRATEGY_REGISTRY, EARN_VAULT, EARN_VAULT_COMPANION } from './config';
+import { encodeFunctionData, Hex, Address as ViemAddress } from 'viem';
+import vaultAbi from '@shared/abis/earn-vault';
+import companionAbi from '@shared/abis/earn-vault-companion';
+import strategyRegistryAbi from '@shared/abis/earn-strategy-registry';
+import strategyAbi from '@shared/abis/earn-strategy';
+import { isSameAddress } from '@shared/utils';
+import { Addresses, Uint } from '@shared/constants';
+import { IPermit2Service, PermitData } from '@services/permit2';
+import { IQuoteService, QuoteRequest } from '@services/quotes';
+import { IProviderService } from '@services/providers';
+import { MULTICALL_CONTRACT } from '@services/providers/utils';
+import { IAllowanceService } from '@services/allowances';
+
+export class EarnService implements IEarnService {
+  constructor(
+    private readonly permit2Service: IPermit2Service,
+    private readonly quoteService: IQuoteService,
+    private readonly providerService: IProviderService,
+    private readonly allowanceservice: IAllowanceService
+  ) {}
+
+  async buildCreatePositionTx({
+    chainId,
+    strategyId,
+    owner,
+    permissions,
+    strategyValidationData,
+    misc,
+    deposit,
+  }: CreateEarnPositionParams): Promise<BuiltTransaction> {
+    let depositInfo: { token: Address; amount: bigint; value: bigint };
+    if ('token' in deposit) {
+      const amount = BigInt(deposit.amount);
+      depositInfo = { token: deposit.token, amount, value: isSameAddress(deposit.token, Addresses.NATIVE_TOKEN) ? amount : 0n };
+    } else {
+      depositInfo = { token: deposit.permitData.token, amount: BigInt(deposit.permitData.amount), value: 0n };
+    }
+
+    const { needsSwap, asset } = await this.checkIfNeedsSwap({ chainId, strategyId, depositToken: depositInfo.token });
+
+    if ('token' in deposit && !needsSwap) {
+      // If don't need to use Permit2, then just call the vault
+      return {
+        to: EARN_VAULT.address(chainId),
+        data: encodeFunctionData({
+          abi: vaultAbi,
+          functionName: 'createPosition',
+          args: [
+            strategyId,
+            depositInfo.token as ViemAddress,
+            depositInfo.amount,
+            owner as ViemAddress,
+            permissions.map(({ operator, permissions }) => ({
+              operator: operator as ViemAddress,
+              permissions: permissions.map(mapPermission),
+            })),
+            strategyValidationData ?? '0x',
+            misc ?? '0x',
+          ],
+        }),
+        value: isSameAddress(depositInfo.token, Addresses.NATIVE_TOKEN) ? depositInfo.amount : undefined,
```

Comment:
I think you already have this here
```suggestion
        value: depositInfo.value,
```